kuma/core/tests/test_views.py

```diff
@@ -1,40 +1,47 @@
-from nose.tools import eq_
+import logging
+from nose.tools import eq_, ok_
 from pyquery import PyQuery as pq
 from soapbox.models import Message
 
-from django.conf import settings
 from django.core import mail
+from django.test import override_settings
+from django.utils.log import AdminEmailHandler
 
 from kuma.core.tests import KumaTestCase
 
 from ..urlresolvers import reverse
 
 
+@override_settings(
+    DEBUG=False,
+    DEBUG_PROPAGATE_EXCEPTIONS=False,
+    ADMINS=(('admin', 'admin@example.com'),))
 class LoggingTests(KumaTestCase):
     urls = 'kuma.core.tests.logging_urls'
-
-    def setUp(self):
-        self.old_logging = settings.LOGGING
-
-    def tearDown(self):
-        settings.LOGGING = self.old_logging
+    logger = logging.getLogger('django.security')
+    suspicous_path = '/en-US/suspicious/'
 
     def test_no_mail_handler(self):
+        old_handlers = self.logger.handlers[:]
         try:
-            response = self.client.get('/en-US/test_exception/')
-            eq_(500, response.status_code)
+            self.logger.handlers = [logging.NullHandler()]
+            response = self.client.get(self.suspicous_path)
+            eq_(response.status_code, 400)
             eq_(0, len(mail.outbox))
-        except:
-            pass
+        finally:
```

Comment:
Are you against ``setup()`` / ``teardown()``, or is there something that requires ``try``/``finally`` instead?